frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -73,6 +73,9 @@ class Worksheet extends React.Component {
             editorEnabled: false, // Whether the editor is actually showing (sometimes lags behind inSourceEditMode)
             showTerminal: false, // Whether the terminal is shown
             focusIndex: -1, // Which worksheet items to be on (-1 is none)
+            newTextIndex: -1, // Which index new text to be on (-1 is at the beginning).
```

Comment:
I'm a bit confused as to how these variables fix the issue. Can you explain how they work?

Reply:
The index stores where the new text/run/schema should be added. Every time an item is focused we update the index.
Before we didn't have this and the default behavior was just remove the item whenever focus changes.

Reply:
@epicfaace can I get a review on this, I think it should be ready to go.

Reply:
Okay, i see what you're doing, but can you update the comments here to make it clearer what's going on and why these state properties are useful? At the moment, it might be hard for someone new looking at the code for the first time to see what they're used for.

Reply:
Thanks, I have added more comments to explain the indices.